core/analysis/text_token_stream.cpp

```diff
@@ -1053,18 +1053,15 @@ bool text_token_stream::next_word() {
       continue;
     }
 
+    // TODO(MBkkt) simdutf::utf8_length_from_utf16
     auto utf8_length = [data = &state_->data](uint32_t begin,
                                               uint32_t end) noexcept {
       uint32_t length = 0;
       while (begin < end) {
         const auto cp = data->char32At(begin);
-
         // icu::UnicodeString::kInvalidUChar is private
-        if (IRS_UNLIKELY(0xFFFF == cp)) {
-          return uint32_t{0};
-        }
-
-        length += utf8_utils::cp_length(cp);
+        IRS_ASSERT(cp != 0xFFFF);
```

Comment:
Can we be sure that `icu::UnicodeString::fromUTF8` always returns valid string? From comment above it looks like it might set icu::UnicodeString::kInvalidUChar in the string in case of conversion error.

Reply:
Fixed